lib/kubernetes-deploy/concurrency.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  module Concurrency
+    MAX_THREADS = 8
+
+    def self.split_across_threads(all_work, &block)
+      return if all_work.empty?
+      raise ArgumentError, "Block of work is required" unless block_given?
+
+      slice_size = ((all_work.length + MAX_THREADS - 1) / MAX_THREADS)
+      threads = []
+      all_work.each_slice(slice_size) do |work_group|
```

Comment:
> otherwise those threads will never report exceptions

That's what I thought that option meant too when I was reading the `Thread` docs, but when I played with it that's not what I saw. It seems like it makes the main thread abort immediately when any other thread aborts, as opposed to when you join the other threads. Even more unexpectedly, it doesn't seem to abort the other child threads immediately. Here's the experiment I did:
```ruby
      threads = []
      Thread.abort_on_exception = true # comment out this line for the other version of experiment
      all_work.each_slice(slice_size) do |work_group|
        thread = Thread.new do
          puts ColorizedString.new("This thing started: #{work_group.map(&:id)}").yellow
          work_group.each(&block)
          puts ColorizedString.new("This thing ran: #{work_group.map(&:id)}").green
          raise FatalDeploymentError, "this thing raised: #{work_group.map(&:id)}" if work_group.any? { |work| work.id.include?("Service/redis") }
        end
        threads << thread
      end
      puts "SLEEPING"
      sleep 10
      puts "JOINING THREADS"
      threads.each(&:join)
```

`Thread.abort_on_exception = false`
![image](https://user-images.githubusercontent.com/4789493/28680547-b6482f48-72c4-11e7-8c87-5670023c70a8.png)


`Thread.abort_on_exception = true`
![image](https://user-images.githubusercontent.com/4789493/28680568-c6cac8da-72c4-11e7-94da-b5a02b16ee51.png)

The former behaviour looks nicer to me. Am I missing something?